### PR TITLE
feat(datasets): upload JSON for dataset examples

### LINKED ISSUE #3655: [datasets][sdk] add the ability to send raw examples
As a user, I want to be able to construct examples as lists and send them AS IS to be used as inputs and outputs. This is not possible with "tabular" data because I can't have conflicting keys

| input | output |
|--------|--------|
| {} | {}|

yields `input.input` nested keys.

Expose a way such that I can send examples as JSONserializable objects so as to avoid this nesting.

### PR DESCRIPTION
**Pull request by RogerHYang**
resolves #3655

### CODE REVIEW

File: src/phoenix/db/insertion/dataset.py [hunk 1]
```diff
@@ -173,9 +178,7 @@ async def add_dataset_examples(
                 created_at=created_at,
             )
         except Exception:
-            logger.exception(
-                f"Fail to insert dataset: {input_keys=}, {output_keys=}, {metadata_keys=}"
-            )
+            logger.exception(f"Fail to insert dataset: {name=}")
```
Comment (mikeldking) [inline src/phoenix/db/insertion/dataset.py:181]:
```suggestion
            logger.exception(f"Failed to insert dataset: {name=}")
```

File: src/phoenix/session/client.py [hunk 2]
```diff
@@ -425,14 +428,97 @@ def download_dataset_examples(
             index_col="example_id",
         )
 
+    def create_examples(
+        self,
+        *,
+        name: str,
```
Comment (mikeldking) [inline src/phoenix/session/client.py:434]: I like the simplicity but it took me a second that name was the name of the dataset and the description was the dataset description as there would be a version too.

**Review by mikeldking — APPROVED** [review]
vnice

File: src/phoenix/session/client.py (hunk 2 above)
Comment (RogerHYang) [reply src/phoenix/session/client.py:434]: Good catch! I'll clean up. Thanks

Comment (RogerHYang) [reply src/phoenix/session/client.py:434]: I don't think version is needed since that's created on the server side.

**Pull request merged**